rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java

```diff
@@ -328,7 +328,8 @@ public void registerBtcTransaction(Transaction rskTx, BtcTransaction btcTx, int
             // That is, build a release transaction and get it in the release transaction set.
             // Otherwise, transfer SBTC to the sender of the BTC
             // The RSK account to update is the one that matches the pubkey "spent" on the first bitcoin tx input
-            if (!provider.getLockWhitelist().isWhitelisted(senderBtcAddress)) {
+            LockWhitelist lockWhitelist = provider.getLockWhitelist();
+            if (!lockWhitelist.isWhitelisted(senderBtcAddress) || totalAmount.isGreaterThan(lockWhitelist.getMaxTransferValue(senderBtcAddress))) {
```

Comment:
This is leaking whitelist logic into `BridgeSupport`. We probably would benefit from `isWhitelistedFor(address, amount)`.